diff -u ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/initres.f source/initres.f
--- ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/initres.f	2014-03-25 14:50:20.000000000 -0400
+++ source/initres.f	2015-07-24 16:12:05.000000000 -0400
@@ -51,13 +51,13 @@
       data acid1  / 'G', 'A', 'V', 'L', 'I', 'S', 'T', 'C', 'C', 'c',
      &              'P', 'F', 'Y', 'y', 'W', 'H', 'U', 'Z', 'D', 'd',
      &              'N', 'E', 'e', 'Q', 'M', 'K', 'k', 'R', 'O', 'B',
-     &              'J', 't', 'f', 'a', 'x', 'n', 'm', 'X' /
+     &              'J', 't', 'f', 'a', 'x', 'n', 'm', 'X', 'r' /
       data acid3  / 'GLY', 'ALA', 'VAL', 'LEU', 'ILE', 'SER', 'THR',
      &              'CYS', 'CYX', 'CYD', 'PRO', 'PHE', 'TYR', 'TYD',
      &              'TRP', 'HIS', 'HID', 'HIE', 'ASP', 'ASH', 'ASN',
      &              'GLU', 'GLH', 'GLN', 'MET', 'LYS', 'LYD', 'ARG',
      &              'ORN', 'AIB', 'PCA', 'H2N', 'FOR', 'ACE', 'COH', 
-     &              'NH2', 'NME', 'UNK' /
+     &              'NH2', 'NME', 'UNK', 'RET' /
 c
 c     supported nucleotide 1-letter and 3-letter codes
 c
@@ -72,37 +72,37 @@
      &               87,   96,  105,  116,  131,  147,  162,  185,
      &              202,  218,  234,  244,  256,  268,  280,  294,
      &              308,  321,  337,  353,  370,  384,  391,    0,
-     &                0,    0,    0,    0,    0,    1 /
+     &                0,    0,    0,    0,    0,    1,    0 /
       data cat   /    2,    8,   16,   28,   42,   56,   66,   78,
      &               88,   97,  106,  117,  132,  148,  163,  186,
      &              203,  219,  235,  245,  257,  269,  281,  295,
      &              309,  322,  338,  354,  371,  385,  392,    0,
-     &                0,    0,    0,    0,    0,    2 /
+     &                0,    0,    0,    0,    0,    2,    0 /
       data ct    /    3,    9,   17,   29,   43,   57,   67,   79,
      &               89,   98,  107,  118,  133,  149,  164,  187,
      &              204,  220,  236,  246,  258,  270,  282,  296,
      &              310,  323,  339,  355,  372,  386,  393,    0,
-     &                0,    0,    0,    0,    0,    3 /
+     &                0,    0,    0,    0,    0,    3,    0 /
       data hnt   /    4,   10,   18,   30,   44,   58,   68,   80,
      &               90,   99,    0,  119,  134,  150,  165,  188,
      &              205,  221,  237,  247,  259,  271,  283,  297,
      &              311,  324,  340,  356,  373,  387,  394,    0,
-     &                0,    0,    0,    0,    0,    4 /
+     &                0,    0,    0,    0,    0,    4,    0 /
       data ot    /    5,   11,   19,   31,   45,   59,   69,   81,
      &               91,  100,  108,  120,  135,  151,  166,  189,
      &              206,  222,  238,  248,  260,  272,  284,  298,
      &              312,  325,  341,  357,  374,  388,  395,    0,
-     &                0,    0,    0,    0,    0,    5 /
+     &                0,    0,    0,    0,    0,    5,    0 /
       data hat   /    6,   12,   20,   32,   46,   60,   70,   82,
      &               92,  101,  109,  121,  136,  152,  167,  190,
      &              207,  223,  239,  249,  261,  273,  285,  299,
      &              313,  326,  342,  358,  375,    0,  396,    0,
-     &                0,    0,    0,    0,    0,    6 /
+     &                0,    0,    0,    0,    0,    6,    0 /
       data cbt   /    0,   13,   21,   33,   47,   61,   71,   83,
      &               93,  102,  110,  122,  137,  153,  168,  191,
      &              208,  224,  240,  250,  262,  274,  286,  300,
      &              314,  327,  343,  359,  376,  389,  397,    0,
-     &                0,    0,    0,    0,    0,    0 /
+     &                0,    0,    0,    0,    0,    0,    0 /
 c
 c     biopolymer types for N-terminal peptide backbone atoms
 c
@@ -110,32 +110,32 @@
      &              451,  457,  463,  471,  477,  483,  489,  495,
      &              501,  507,  513,  519,  525,  531,  537,  543,
      &              549,  555,  561,  567,  573,  579,  391,  762,
-     &                0,    0,    0,    0,    0,  403 /
+     &                0,    0,    0,    0,    0,  403,    0 /
       data can   /  404,  410,  416,  422,  428,  434,  440,  446,
      &              452,  458,  464,  472,  478,  484,  490,  496,
      &              502,  508,  514,  520,  526,  532,  538,  544,
      &              550,  556,  562,  568,  574,  580,  392,    0,
-     &                0,  767,    0,    0,    0,  404 /
+     &                0,  767,    0,    0,    0,  404,    0 /
       data cn    /  405,  411,  417,  423,  429,  435,  441,  447,
      &              453,  459,  465,  473,  479,  485,  491,  497,
      &              503,  509,  515,  521,  527,  533,  539,  545,
      &              551,  557,  563,  569,  575,  581,  393,    0,
-     &              764,  769,    0,    0,    0,  405 /
+     &              764,  769,    0,    0,    0,  405,    0 /
       data hnn   /  406,  412,  418,  424,  430,  436,  442,  448,
      &              454,  460,  466,  474,  480,  486,  492,  498,
      &              504,  510,  516,  522,  528,  534,  540,  546,
      &              552,  558,  564,  570,  576,  582,  394,  763,
-     &                0,    0,    0,    0,    0,  406 /
+     &                0,    0,    0,    0,    0,  406,    0 /
       data on    /  407,  413,  419,  425,  431,  437,  443,  449,
      &              455,  461,  467,  475,  481,  487,  493,  499,
      &              505,  511,  517,  523,  529,  535,  541,  547,
      &              553,  559,  565,  571,  577,  583,  395,    0,
-     &              766,  770,    0,    0,    0,  407 /
+     &              766,  770,    0,    0,    0,  407,    0 /
       data han   /  408,  414,  420,  426,  432,  438,  444,  450,
      &              456,  462,  468,  476,  482,  488,  494,  500,
      &              506,  512,  518,  524,  530,  536,  542,  548,
      &              554,  560,  566,  572,  578,    0,  396,    0,
-     &              765,  768,    0,    0,    0,  408 /
+     &              765,  768,    0,    0,    0,  408,    0 /
 c
 c     biopolymer types for C-terminal peptide backbone atoms
 c
@@ -143,32 +143,32 @@
      &              632,  638,  644,  649,  655,  661,  667,  673,
      &              679,  685,  691,  697,  703,  709,  715,  721,
      &              727,  733,  739,  745,  751,  757,    0,    0,
-     &                0,    0,  773,  775,  777,  584 /
+     &                0,    0,  773,  775,  777,  584,    0 /
       data cac   /  585,  591,  597,  603,  609,  615,  621,  627,
      &              633,  639,  645,  650,  656,  662,  668,  674,
      &              680,  686,  692,  698,  704,  710,  716,  722,
      &              728,  734,  740,  746,  752,  758,    0,    0,
-     &                0,    0,    0,    0,  779,  585 /
+     &                0,    0,    0,    0,  779,  585,    0 /
       data cc    /  586,  592,  598,  604,  610,  616,  622,  628,
      &              634,  640,  646,  651,  657,  663,  669,  675,
      &              681,  687,  693,  699,  705,  711,  717,  723,
      &              729,  735,  741,  747,  753,  759,    0,    0,
-     &                0,    0,  771,    0,    0,  586 /
+     &                0,    0,  771,    0,    0,  586,    0 /
       data hnc   /  587,  593,  599,  605,  611,  617,  623,  629,
      &              635,  641,    0,  652,  658,  664,  670,  676,
      &              682,  688,  694,  700,  706,  712,  718,  724,
      &              730,  736,  742,  748,  754,  760,    0,    0,
-     &                0,    0,  774,  776,  778,  587 /
+     &                0,    0,  774,  776,  778,  587,    0 /
       data oc    /  588,  594,  600,  606,  612,  618,  624,  630,
      &              636,  642,  647,  653,  659,  665,  671,  677,
      &              683,  689,  695,  701,  707,  713,  719,  725,
      &              731,  737,  743,  749,  755,  761,    0,    0,
-     &                0,    0,  772,    0,    0,  588 /
+     &                0,    0,  772,    0,    0,  588,    0 /
       data hac   /  589,  595,  601,  607,  613,  619,  625,  631,
      &              637,  643,  648,  654,  660,  666,  672,  678,
      &              684,  690,  696,  702,  708,  714,  720,  726,
      &              732,  738,  744,  750,  756,    0,    0,    0,
-     &                0,    0,    0,    0,  780,  589 /
+     &                0,    0,    0,    0,  780,  589,    0 /
 c
 c     biopolymer types for nucleotide phosphate and sugar atoms
 c
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/initres.o and source/initres.o differ
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/libtinker.a and source/libtinker.a differ
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/nucleic.o and source/nucleic.o differ
Only in source/: patch1
diff -u ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/pdbxyz.f source/pdbxyz.f
--- ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/pdbxyz.f	2014-03-25 14:50:21.000000000 -0400
+++ source/pdbxyz.f	2015-07-24 16:12:05.000000000 -0400
@@ -2015,19 +2015,25 @@
       include 'pdb.i'
       integer i
 c
+c Retinal, acetate 
+c     
+      logical first, primo
+c
 c
 c     find water molecules and ions in PDB HETATM records
 c
       n = n + 1
       i = 0
+      first = .false.
+      primo = .false.
       do while (i .lt. npdb)
          i = i + 1
          if (pdbtyp(i) .eq. 'HETATM') then
-            if (resnam(i) .eq. 'HOH') then
+            if ((resnam(i) .eq. 'HOH') .or. (resnam(i) .eq. 'WAT')) then
                if (atmnam(i) .eq. ' O  ') then
                   call oldatm (i,2001,0,0)
-                  if (atmnam(i+1).eq.' H  ' .and.
-     &                atmnam(i+2).eq.' H  ') then
+                  if ((atmnam(i+1).eq.' H ') .and.
+     &                (atmnam(i+2).eq.' H ')) then
                      call oldatm (i+1,2002,n-1,0)
                      call oldatm (i+2,2002,n-2,0)
                      i = i + 2
@@ -2038,6 +2044,13 @@
      &                               n-3,120.0d0,0)
                   end if
                end if
+c
+c Retinal
+c
+            else if ((resnam(i) .eq. 'RET') .and. (.not.first)) then
+               call buildret(i,first)
+            else if ((resnam(i) .eq. 'ACI') .and. (.not.primo)) then
+               call buildaci(i,primo)
             else if (resnam(i) .eq. 'NA ') then
                call oldatm (i,2003,0,0)
             else if (resnam(i) .eq. 'K  ') then
@@ -2047,7 +2060,7 @@
             else if (resnam(i) .eq. 'CA ') then
                call oldatm (i,2006,0,0)
             else if (resnam(i) .eq. 'CL ') then
-               call oldatm (i,2007,0,0)
+               call oldatm (i,2015,0,0)
             end if
          end if
       end do
@@ -2055,6 +2068,293 @@
       return
       end
 c
+c Retinal
+c
+c     #################################################################
+c     ##                                                             ##
+c     ##  subroutine buildret  --  coordinates of retinal            ##
+c     ##                                                             ##
+c     #################################################################
+c
+c
+c     "buildret" translates retinal in Protein Data Bank format to a
+c     Cartesian coordinate file and sequence file
+c
+c
+      subroutine buildret(i,stop)
+      implicit none
+      include 'sizes.i'
+      include 'atoms.i'
+      include 'atmtyp.i'
+      include 'pdb.i'
+      integer i,j,k,symb,cont,hncont,a,b
+      character*2 jstring
+      logical atm,stop
+      real*8 x2,y2,z2,dist
+      real*8 xdih(4),ydih(4),zdih(4),rAB,rAC,rAD,rBC,rBD,rCD,qterm
+      real*8 pterm,diha,dihn
+c
+c
+c     decide if it is a sp2 or sp3 atom, based on the numeration
+c
+      j = 1
+      atm = .false.
+      do while ((j .le. 20) .and. (.not.atm))
+         if (j .lt. 10) then
+            write(jstring,'(i1)') j
+            atm = (atmnam(i) .eq. ' C'//jstring//' ')
+         else
+            write(jstring,'(i2)') j
+            atm = (atmnam(i) .eq. ' C'//jstring)
+         end if
+         if (atm) then
+            if (j .eq. 1) call oldatm(i,2013,0,0)
+            if ((j .gt. 1) .and. (j .le. 4)) call oldatm(i,2013,n-1,0)
+            if ((j .ge. 5) .and. (j .lt. 16)) call oldatm(i,2012,n-1,0)
+            if (j .eq. 6) call addbond(n-1,n-6)
+            if ((j .eq. 16) .or. (j .eq. 17)) 
+     &                     call oldatm(i,2013,n-j+1,0)
+            if (j .ge. 18) call oldatm(i,2013,n-(13-3*(j-18)),0)
+         end if
+         j = j + 1
+      end do
+c
+c     after carbon atoms, H Ret are added
+c
+      if (j .eq. 21) then
+         stop = .true.
+         write(6,*) ' '
+         write(6,*) 'WARNING: Retinal Moiety Detected! '
+         write(6,*) 'The retinal H atoms might be in the wrong position'
+         write(6,*) ' '
+         k = 1
+c
+c     C2 H atoms
+c
+         call newatm (0,2014,n-19,1.08d0,n-20,109.5d0,n-15,120.0d0,0)
+         call newatm (0,2014,n-20,1.08d0,n-21,109.5d0,n-16,-120.0d0,0)
+         k = k + 2
+c
+c    C3 H atoms
+c
+         call newatm (0,2014,n-20,1.08d0,n-19,109.5d0,n-18,60.0d0,0)
+         call newatm (0,2014,n-21,1.08d0,n-20,109.5d0,n-19,180.0d0,0) 
+         k = k + 2
+c
+c    C4 H atoms
+c
+         call newatm (0,2014,n-21,1.08d0,n-20,109.5d0,n-19,120.0d0,0)
+         call newatm (0,2014,n-22,1.08d0,n-21,109.5d0,n-20,-120.0d0,0) 
+         k = k + 2
+c
+c    cycle for C7 and C8 H atoms
+c
+         do while (k .le. 8)
+            call newatm (0,2014,n-20,1.08d0,n-19,120.0d0,n-18,0.0d0,0)
+            k = k + 1
+         end do
+c
+c    cycle for C10 and C11 H atoms
+c
+         do while (k .le. 10)
+            call newatm (0,2014,n-19,1.08d0,n-20,120.0d0,n-21,0.0d0,0)
+            k = k + 1
+         end do
+c
+c    H atom on C12
+c
+         call newatm (0,2014,n-19,1.08d0,n-18,120.0d0,n-17,0.0d0,0)
+         k = k + 1
+c
+c    C14
+c
+
+      xdih(1) = x(n-20)
+      xdih(2) = x(n-19)
+      xdih(3) = x(n-18)
+      xdih(4) = x(n-17)
+      ydih(1) = y(n-20)
+      ydih(2) = y(n-19)
+      ydih(3) = y(n-18)
+      ydih(4) = y(n-17)
+      zdih(1) = z(n-20)
+      zdih(2) = z(n-19)
+      zdih(3) = z(n-18)
+      zdih(4) = z(n-17)
+     
+      rCD=sqrt((xdih(4)-xdih(3))**2 + (ydih(4)-ydih(3))**2 +
+     &(zdih(4)-zdih(3))**2)
+      rBC=sqrt((xdih(3)-xdih(2))**2 + (ydih(3)-ydih(2))**2 +
+     &(zdih(3)-zdih(2))**2)
+      rBD=sqrt((xdih(4)-xdih(2))**2 + (ydih(4)-ydih(2))**2 +
+     &(zdih(4)-zdih(2))**2)
+
+      rAB=sqrt((xdih(2)-xdih(1))**2 + (ydih(2)-ydih(1))**2 +
+     &(zdih(2)-zdih(1))**2)
+      rAD=sqrt((xdih(4)-xdih(1))**2 + (ydih(4)-ydih(1))**2 +
+     &(zdih(4)-zdih(1))**2)
+      rAC=sqrt((xdih(3)-xdih(1))**2 + (ydih(3)-ydih(1))**2 +
+     &(zdih(3)-zdih(1))**2)
+
+      pterm=(rCD**2)*(rBC**2+rAB**2-rAC**2)+(rBC**2)*(-rBC**2+
+     &rAB**2+rAC**2)+(rBD**2)*(rBC**2-rAB**2+rAC**2)-
+     &2*(rBC**2)*(rAD**2)
+      qterm=(rCD+rBC+rBD)*(rCD+rBC-rBD)*(rCD-rBC+rBD)*(-rCD+rBC+rBD)*
+     &(rBC+rAB+rAC)*(rBC+rAB-rAC)*(rBC-rAB+rAC)*(-rBC+rAB+
+     &rAC)
+
+      diha=acos(pterm/sqrt(qterm))*(180/3.14159)
+
+      if (abs(diha) .gt. 90.0) then
+         call newatm (0,2014,n-18,1.08d0,n-19,120.0d0,n-20,0.0d0,0)
+         dihn = 0.0d0
+      else
+         call newatm (0,2014,n-18,1.08d0,n-19,120.0d0,n-20,180.0d0,0)
+         dihn = 180.0d0
+      end if
+      k = k + 1
+      
+c
+c    C15
+c
+            call newatm (0,2014,n-18,1.08d0,n-19,120.0d0,n-20,0.0d0,0)
+            k = k + 1
+c
+c    H atoms on C16
+c
+         call newatm (0,2014,n-18,1.08d0,n-33,109.5d0,n-32,0.0d0,0)
+         call newatm (0,2014,n-19,1.08d0,n-34,109.5d0,n-33,
+     &                      -120.0d0,0)
+         call newatm (0,2014,n-20,1.08d0,n-35,109.5d0,n-34,120.0d0,0)
+         k = k + 1
+c
+c    H atoms on C17
+c
+         call newatm (0,2014,n-20,1.08d0,n-36,109.5d0,n-35,0.0d0,0)
+         call newatm (0,2014,n-21,1.08d0,n-37,109.5d0,n-36,
+     &                      -120.0d0,0)
+         call newatm (0,2014,n-22,1.08d0,n-38,109.5d0,n-37,120.0d0,0)
+         k = k + 1
+c
+c    H atoms on C18
+c
+         call newatm (0,2014,n-22,1.08d0,n-35,109.5d0,n-34,180.0d0,0)
+         call newatm (0,2014,n-23,1.08d0,n-36,109.5d0,n-35,
+     &                      -60.0d0,0)
+         call newatm (0,2014,n-24,1.08d0,n-37,109.5d0,n-36,60.0d0,0)
+         k = k + 1
+c
+c    H atoms on C19
+c
+         call newatm (0,2014,n-24,1.08d0,n-34,109.5d0,n-33,150.0d0,0)
+         call newatm (0,2014,n-25,1.08d0,n-35,109.5d0,n-34,
+     &                      -90.0d0,0)
+         call newatm (0,2014,n-26,1.08d0,n-36,109.5d0,n-35,30.0d0,0)
+         k = k + 1
+c
+c    H atoms on C20
+c
+         call newatm (0,2014,n-26,1.08d0,n-33,109.5d0,n-32,150.0d0,0)
+         call newatm (0,2014,n-27,1.08d0,n-34,109.5d0,n-33,
+     &                      -90.0d0,0)
+         call newatm (0,2014,n-28,1.08d0,n-35,109.5d0,n-34,30.0d0,0)
+c
+c    In case of H Ret in PDB file, use their PDB coordinates
+c
+         if (resnam(i+1) .eq. 'RET') then
+            a = i+1
+            b = i+28
+            do cont = a, b
+              x(cont+2) = xpdb(cont)
+              y(cont+2) = ypdb(cont)
+              z(cont+2) = zpdb(cont)
+            end do
+            hncont = cont-1
+         end if
+c
+c    Lysine search and connection
+c    C15 coordinates are current position minus 34
+c
+         k = 1 
+         atm = .false.
+c
+c    loop over all atoms, finishing when the PSB Lys has been found
+c
+         do while ((k .le. n) .and. .not.atm)
+            symb = index(name(k),'N3')
+c
+c    compute the distance from C15 only for N3 Tinker atom name
+c
+            if (symb .ne. 0) then
+               x2 = (x(n-34)-x(k))**2
+               y2 = (y(n-34)-y(k))**2
+               z2 = (z(n-34)-z(k))**2
+               dist = sqrt(x2+y2+z2)
+c
+c    if the distance is lower than 1.8 A, 2 HZ atoms are deleted,
+c    the C15-N connection is made and the coordinates for the
+c    for the 3rd HZ are modified according to C15-N double bond
+c
+               if (dist .le. 1.8d0) then
+                  do cont = 1, 2
+                     call delete(k+9)
+                  end do
+                  call addbond(n-34,k)
+                  if (resnam(k).eq.'LYS') then
+                     call xyzatm(k+9,k,1.04d0,n-34,120.0d0,n-35,dihn,0)
+                     if (resnam(i+1) .eq. 'RET') then
+                        x(k+9) = xpdb(k+7)
+                        y(k+9) = ypdb(k+7)
+                        z(k+9) = zpdb(k+7)
+                     end if
+                  end if
+                  atm = .true. 
+               end if
+            end if
+            k = k + 1
+         end do
+      end if
+      return
+      end
+
+c Free acetate ion
+c
+c     #################################################################
+c     ##                                                             ##
+c     ##  subroutine buildaci  --  coordinates of CH3COO-            ##
+c     ##                                                             ##
+c     #################################################################
+c
+c
+c     "buildaci" translates acetate in Protein Data Bank format to a
+c     Cartesian coordinate file and sequence file
+c
+c
+      subroutine buildaci(i,primo)
+      implicit none
+      include 'sizes.i'
+      include 'atoms.i'
+      include 'atmtyp.i'
+      include 'pdb.i'
+      integer i
+      logical primo
+c
+c
+c     adds all the atom in the usual order for acetate
+c
+      
+      call oldatm (i,340,0,0)
+      call oldatm (i+1,341,n-1,0)
+      call oldatm (i+2,341,n-2,0)
+      call oldatm (i+3,341,n-3,0)
+      call oldatm (i+4,342,n-4,0)
+      call oldatm (i+5,343,n-1,0)
+      call oldatm (i+6,343,n-2,0)
+      if (.not. primo) primo = .true.
+      return 
+      end
+
+c
 c
 c     ############################################################
 c     ##                                                        ##
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/pdbxyz.o and source/pdbxyz.o differ
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/protein.o and source/protein.o differ
diff -u ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/readpdb.f source/readpdb.f
--- ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/readpdb.f	2014-03-25 14:50:21.000000000 -0400
+++ source/readpdb.f	2015-07-24 16:12:05.000000000 -0400
@@ -860,6 +860,13 @@
          if (atmname .eq. ' HB1')  atmname = ' HB3'
          if (atmname .eq. '2HB ')  atmname = ' HB3'
 c
+c     protonated aspartic acid residue  (ASH)
+c
+      else if (resname .eq. 'ASH') then
+         if (atmname .eq. '1HB ')  atmname = ' HB2'
+         if (atmname .eq. ' HB1')  atmname = ' HB3'
+         if (atmname .eq. '2HB ')  atmname = ' HB3'
+c
 c     asparagine residue  (ASN)
 c
       else if (resname .eq. 'ASN') then
@@ -883,6 +890,17 @@
          if (atmname .eq. ' HG1')  atmname = ' HG3'
          if (atmname .eq. '2HG ')  atmname = ' HG3'
 c
+c     protonated glutamic acid residue  (GLH)
+c
+      else if (resname .eq. 'GLH') then
+         if (atmname .eq. '1HB ')  atmname = ' HB2'
+         if (atmname .eq. ' HB1')  atmname = ' HB3'
+         if (atmname .eq. '2HB ')  atmname = ' HB3'
+         if (atmname .eq. '1HG ')  atmname = ' HG2'
+         if (atmname .eq. ' HG1')  atmname = ' HG3'
+         if (atmname .eq. '2HG ')  atmname = ' HG3'
+
+c
 c     glutamine residue  (GLN)
 c
       else if (resname .eq. 'GLN') then
@@ -934,6 +952,26 @@
          if (atmname .eq. '3HZ ')  atmname = ' HZ3'
          if (atmname .eq. 'HNZ3')  atmname = ' HZ3'
 c
+c     neutral lysine residue  (LYD)
+c
+      else if (resname .eq. 'LYD') then
+         if (atmname .eq. '1HB ')  atmname = ' HB2'
+         if (atmname .eq. ' HB1')  atmname = ' HB3'
+         if (atmname .eq. '2HB ')  atmname = ' HB3'
+         if (atmname .eq. '1HG ')  atmname = ' HG2'
+         if (atmname .eq. ' HG1')  atmname = ' HG3'
+         if (atmname .eq. '2HG ')  atmname = ' HG3'
+         if (atmname .eq. '1HD ')  atmname = ' HD2'
+         if (atmname .eq. ' HD1')  atmname = ' HD3'
+         if (atmname .eq. '2HD ')  atmname = ' HD3'
+         if (atmname .eq. '1HE ')  atmname = ' HE2'
+         if (atmname .eq. ' HE1')  atmname = ' HE3'
+         if (atmname .eq. '2HE ')  atmname = ' HE3'
+         if (atmname .eq. '1HZ ')  atmname = ' HZ1'
+         if (atmname .eq. 'HNZ1')  atmname = ' HZ1'
+         if (atmname .eq. '2HZ ')  atmname = ' HZ2'
+         if (atmname .eq. 'HNZ2')  atmname = ' HZ2'
+c
 c     arginine residue  (ARG)
 c
       else if (resname .eq. 'ARG') then
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/readpdb.o and source/readpdb.o differ
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/readseq.o and source/readseq.o differ
diff -u ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/sizes.i source/sizes.i
--- ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/sizes.i	2014-03-25 14:50:21.000000000 -0400
+++ source/sizes.i	2015-07-24 16:12:05.000000000 -0400
@@ -89,7 +89,7 @@
       parameter (maxbio=10000)
       parameter (maxres=10000)
       parameter (maxele=112)
-      parameter (maxamino=38)
+      parameter (maxamino=39)
       parameter (maxnuc=12)
       parameter (maxbnd=2*maxatm)
       parameter (maxang=4*maxatm)
diff -u ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/xyzpdb.f source/xyzpdb.f
--- ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/xyzpdb.f	2014-03-25 14:50:21.000000000 -0400
+++ source/xyzpdb.f	2015-07-25 10:47:55.000000000 -0400
@@ -125,6 +125,12 @@
       character*7, allocatable :: restyp(:)
       character*120 seqfile
 c
+c     New variables for retinal 
+c
+      logical caret
+      character*1 conto
+      character*2 conto2
+c
 c
 c     initialize number of PDB atoms and atom mapping
 c
@@ -229,7 +235,7 @@
                k = kmol(j)
                atmname = ' '//name(k)
                if (water(i)) then
-                  resname = 'HOH'
+                  resname = 'WAT'
                else
                   justify = 0
                   call numeral (type(k),resname,justify)
@@ -254,6 +260,10 @@
 c     find the amide nitrogens and other peptide backbone atoms
 c
       m = 1
+c
+c    Variables for retinal recognition
+c
+      caret = .false.
       do i = 1, n
          if (restyp(m) .eq. 'PEPTIDE') then
             resname = amino(seqtyp(m))
@@ -363,6 +373,16 @@
                      m = m + 1
                   end if
                end if
+c
+c   First carbon of the RET residue
+c               
+            else if (resname .eq. 'RET' .and. name(i) .eq. 'C3R') then
+              if (.not.caret) then
+                 cai(m) = i
+                 caret = .true.
+              else
+                 m = m + 1
+              end if
             else
                if (atomic(i) .eq. 7) then
                   obone = .false.
@@ -518,12 +538,29 @@
                else if (resname .eq. 'NME') then
                   call pdbatm (' N  ',resname,i,ni(i))
                   call pdbatm (' CH3',resname,i,cai(i))
+c
+c Put all the heavy atoms, followed by the hydrogen for the RET residue
+c
+               else if (resname .eq. 'RET') then
+                  do j = 0, 8
+                     write(conto,'(I1)') j+1
+                     call pdbatm (' C'//conto//' ',resname,i,cai(i)+j)
+                  end do
+                  do j = 9, 19
+                     write(conto2,'(I2)') j+1
+                     call pdbatm (' C'//conto2,resname,i,cai(i)+j)
+                  end do
+                  do j = 20, 47
+                     write(conto2,'(I2)') j+1
+                     call pdbatm (' H'//conto2,resname,i,cai(i)+j)
+                  end do
                else
                   call pdbatm (' N  ',resname,i,ni(i))
                   call pdbatm (' CA ',resname,i,cai(i))
                   call pdbatm (' C  ',resname,i,ci(i))
                   call pdbatm (' O  ',resname,i,oi(i))
                end if
+               call getprohbone (resname,i,m,ni(i),cai(i),cbi)
                call getside (resname,i,ci(i),cai(i),cbi)
                if (resname.eq.'CYS' .or. resname.eq.'CYX') then
                   resname = 'CYS'
@@ -617,9 +654,9 @@
                   justify = 0
                   call numeral (type(k),resname,justify)
                   if (water(i)) then
-                     if (atmnum .eq. 1)  atmname = ' H  '
-                     if (atmnum .eq. 8)  atmname = ' O  '
-                     resname = 'HOH'
+                     if (atmnum .eq. 1)  atmname = ' HT '
+                     if (atmnum .eq. 8)  atmname = ' OT '
+                     resname = 'WAT'
                   else if (atmnum .eq. 11) then
                      atmname = 'NA  '
                      resname = ' NA'
@@ -741,6 +778,7 @@
       if (resname .eq. 'COH')  return
       if (resname .eq. 'NH2')  return
       if (resname .eq. 'NME')  return
+      if (resname .eq. 'RET')  return
 c
 c     find the beta carbon atom for the current residue
 c
@@ -973,6 +1011,11 @@
          call pdbatm (' NH1',resname,ires,cbi+5)
          call pdbatm (' NH2',resname,ires,cbi+6)
 c
+c    retinal (RET)
+c
+      else if (resname .eq. 'RET') then
+         call pdbatm (' C2R',resname,ires,cbi+1)
+c
 c     ornithine residue  (ORN)
 c
       else if (resname .eq. 'ORN') then
@@ -1003,17 +1046,18 @@
 c
 c     ################################################################
 c     ##                                                            ##
-c     ##  subroutine getproh  --  extract the amino acid hydrogens  ##
+c     ##  subroutine getprohbone  --  backbone amino acid H atoms   ##
 c     ##                                                            ##
 c     ################################################################
 c
+c     this was added to change the hydrogen order in the pdb (melaccios)
 c
 c     "getproh" finds the hydrogen atoms for a single amino acid
 c     residue and copies the names and coordinates to the Protein
 c     Data Bank file
 c
 c
-      subroutine getproh (resname,ires,jchain,ni,cai,cbi)
+      subroutine getprohbone (resname,ires,jchain,ni,cai,cbi)
       implicit none
       include 'sizes.i'
       include 'atmtyp.i'
@@ -1024,26 +1068,26 @@
       integer i,nh,hca
       integer ires,jchain
       integer ni,cai,cbi
-      logical allatom
       character*3 resname
       character*4 atmname
+      logical allatom
 c
 c
 c     get any amide hydrogen atoms for non-N-terminal residues
 c
-      if (ires.ne.ichain(1,jchain) .or. n12(ni).ne.4) then
+      if (ires .ne. ichain(1,jchain)) then
          if (resname .ne. 'PRO') then
             do i = 1, n
                if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
                   if (resname .eq. 'COH') then
                      call pdbatm (' HO ',resname,ires,i)
                   else if (resname .eq. 'NH2') then
-                     call pdbatm (' H1 ',resname,ires,i)
-                     call pdbatm (' H2 ',resname,ires,i+1)
+                     call pdbatm (' 1H ',resname,ires,i)
+                     call pdbatm (' 2H ',resname,ires,i+1)
                   else
                      call pdbatm (' H  ',resname,ires,i)
                   end if
-                  goto 10
+                  goto 11
                end if
             end do
          end if
@@ -1057,12 +1101,12 @@
                if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
                   nh = nh + 1
                   if (nh .eq. 1) then
-                     atmname = ' H1 '
+                     atmname = ' 1H '
                   else if (nh .eq. 2) then
-                     atmname = ' H2 '
+                     atmname = ' 2H '
                   end if
                   call pdbatm (atmname,resname,ires,i)
-                  if (nh .eq. 2)  goto 10
+                  if (nh .eq. 2)  goto 11
                end if
             end do
          else if (resname .eq. 'PCA') then
@@ -1070,7 +1114,7 @@
                if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
                   atmname = ' H  '
                   call pdbatm (atmname,resname,ires,i)
-                  goto 10
+                  goto 11
                end if
             end do
          else
@@ -1079,19 +1123,133 @@
                if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
                   nh = nh + 1
                   if (nh .eq. 1) then
-                     atmname = ' H1 '
+                     atmname = ' 1H '
                   else if (nh .eq. 2) then
-                     atmname = ' H2 '
+                     atmname = ' 2H '
                   else if (nh .eq. 3) then
-                     atmname = ' H3 '
+                     atmname = ' 3H '
                   end if
                   call pdbatm (atmname,resname,ires,i)
-                  if (nh .eq. 3)  goto 10
+                  if (nh .eq. 3)  goto 11
                end if
             end do
          end if
       end if
-   10 continue
+   11 continue
+c
+c     get the alpha hydrogen atom for the current residue
+c
+      hca = 0
+      do i = 1, n
+         if (atomic(i).eq.1 .and. i12(1,i).eq.cai) then
+            hca = i
+            if (resname .eq. 'GLY') then
+               atmname = ' 1HA'
+            else if (resname .eq. 'FOR') then
+               atmname = ' H  '
+            else if (resname .eq. 'ACE') then
+               atmname = '1H '
+            else if (resname .eq. 'NME') then
+               atmname = '1H '
+            else
+               atmname = ' HA '
+            end if
+            call pdbatm (atmname,resname,ires,i)
+            goto 21
+         end if
+      end do
+   21 return
+      end     
+c
+c
+c     ################################################################
+c     ##                                                            ##
+c     ##  subroutine getproh  --  extract the amino acid hydrogens  ##
+c     ##                                                            ##
+c     ################################################################
+c
+c
+c     "getproh" finds the hydrogen atoms for a single amino acid
+c     residue and copies the names and coordinates to the Protein
+c     Data Bank file
+c
+c
+      subroutine getproh (resname,ires,jchain,ni,cai,cbi)
+      implicit none
+      include 'sizes.i'
+      include 'atmtyp.i'
+      include 'atoms.i'
+      include 'couple.i'
+      include 'sequen.i'
+      integer i,nh,hca
+      integer ires,jchain
+      integer ni,cai,cbi
+      logical allatom
+      character*3 resname
+      character*4 atmname
+cc
+cc
+cc     get any amide hydrogen atoms for non-N-terminal residues
+cc
+c      if (ires.ne.ichain(1,jchain) .or. n12(ni).ne.4) then
+c         if (resname .ne. 'PRO') then
+c           do i = 1, n
+c               if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
+c                  if (resname .eq. 'NH2') then
+c                     call pdbatm (' H1 ',resname,ires,i)
+c                     call pdbatm (' H2 ',resname,ires,i+1)
+c                  else
+c                     call pdbatm (' H  ',resname,ires,i)
+c                  end if
+c                  goto 12
+c               end if
+c            end do
+c         end if
+cc
+ccc     get any amide hydrogen atoms for N-terminal residues
+ccc
+c      else
+c         if (resname .eq. 'PRO') then
+c            nh = 0
+c            do i = 1, n
+c               if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
+c                  nh = nh + 1
+c                  if (nh .eq. 1) then
+c                     atmname = ' H1 '
+c                  else if (nh .eq. 2) then
+c                     atmname = ' H2 '
+c                  end if
+c                  call pdbatm (atmname,resname,ires,i)
+c                  if (nh .eq. 2)  goto 12
+c               end if
+c            end do
+c         else if (resname .eq. 'PCA') then
+c            do i = 1, n
+c               if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
+c                  atmname = ' H  '
+c                  call pdbatm (atmname,resname,ires,i)
+c                  goto 12
+c               end if
+c            end do
+c         else
+c            nh = 0
+c            do i = 1, n
+c               if (atomic(i).eq.1 .and. i12(1,i).eq.ni) then
+c                  nh = nh + 1
+c                  if (nh .eq. 1) then
+c                     atmname = ' H1 '
+c                  else if (nh .eq. 2) then
+c                     atmname = ' H2 '
+c                  else if (nh .eq. 3) then
+c                     atmname = ' H3 '
+c                  end if
+c                  call pdbatm (atmname,resname,ires,i)
+c                  if (nh .eq. 3)  goto 12
+c               end if
+c            end do
+c         end if
+c      end if
+c   12 continue
 c
 c     get the alpha hydrogen atom for the current residue
 c
@@ -1110,7 +1268,7 @@
             else
                atmname = ' HA '
             end if
-            call pdbatm (atmname,resname,ires,i)
+c            call pdbatm (atmname,resname,ires,i)
             goto 20
          end if
       end do
@@ -1220,8 +1378,8 @@
             call pdbatm (' HB2',resname,ires,hca+3)
             call pdbatm (' HB3',resname,ires,hca+4)
             call pdbatm (' HG ',resname,ires,hca+5)
-         else if (biotyp(86) .ne. 0) then
-            call pdbatm (' HG ',resname,ires,cbi+2)
+c         else if (biotyp(86) .ne. 0) then
+c            call pdbatm (' HG ',resname,ires,cbi+2)
          end if
 c
 c     cystine residue  (CYX)
@@ -1263,12 +1421,12 @@
             call pdbatm (' HE1',resname,ires,hca+12)
             call pdbatm (' HE2',resname,ires,hca+13)
             call pdbatm (' HZ ',resname,ires,hca+14)
-         else if (biotyp(126) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+7)
-            call pdbatm (' HD2',resname,ires,cbi+8)
-            call pdbatm (' HE1',resname,ires,cbi+9)
-            call pdbatm (' HE2',resname,ires,cbi+10)
-            call pdbatm (' HZ ',resname,ires,cbi+11)
+c         else if (biotyp(126) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+7)
+c            call pdbatm (' HD2',resname,ires,cbi+8)
+c            call pdbatm (' HE1',resname,ires,cbi+9)
+c            call pdbatm (' HE2',resname,ires,cbi+10)
+c            call pdbatm (' HZ ',resname,ires,cbi+11)
          end if
 c
 c     tyrosine residue  (TYR)
@@ -1282,12 +1440,12 @@
             call pdbatm (' HE1',resname,ires,hca+13)
             call pdbatm (' HE2',resname,ires,hca+14)
             call pdbatm (' HH ',resname,ires,hca+15)
-         else if (biotyp(141) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+8)
-            call pdbatm (' HD2',resname,ires,cbi+9)
-            call pdbatm (' HE1',resname,ires,cbi+10)
-            call pdbatm (' HE2',resname,ires,cbi+11)
-            call pdbatm (' HH ',resname,ires,cbi+12)
+c         else if (biotyp(141) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+8)
+c            call pdbatm (' HD2',resname,ires,cbi+9)
+c            call pdbatm (' HE1',resname,ires,cbi+10)
+c            call pdbatm (' HE2',resname,ires,cbi+11)
+c            call pdbatm (' HH ',resname,ires,cbi+12)
          else
             call pdbatm (' HH ',resname,ires,cbi+8)
          end if
@@ -1302,11 +1460,11 @@
             call pdbatm (' HD2',resname,ires,hca+12)
             call pdbatm (' HE1',resname,ires,hca+13)
             call pdbatm (' HE2',resname,ires,hca+14)
-         else if (biotyp(141) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+8)
-            call pdbatm (' HD2',resname,ires,cbi+9)
-            call pdbatm (' HE1',resname,ires,cbi+10)
-            call pdbatm (' HE2',resname,ires,cbi+11)
+c         else if (biotyp(141) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+8)
+c            call pdbatm (' HD2',resname,ires,cbi+9)
+c            call pdbatm (' HE1',resname,ires,cbi+10)
+c            call pdbatm (' HE2',resname,ires,cbi+11)
          end if
 c
 c     tryptophan residue  (TRP)
@@ -1321,13 +1479,13 @@
             call pdbatm (' HZ2',resname,ires,hca+16)
             call pdbatm (' HZ3',resname,ires,hca+17)
             call pdbatm (' HH2',resname,ires,hca+18)
-         else if (biotyp(172) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+10)
-            call pdbatm (' HE1',resname,ires,cbi+11)
-            call pdbatm (' HE3',resname,ires,cbi+12)
-            call pdbatm (' HZ2',resname,ires,cbi+13)
-            call pdbatm (' HZ3',resname,ires,cbi+14)
-            call pdbatm (' HH2',resname,ires,cbi+15)
+c         else if (biotyp(172) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+10)
+c            call pdbatm (' HE1',resname,ires,cbi+11)
+c            call pdbatm (' HE3',resname,ires,cbi+12)
+c            call pdbatm (' HZ2',resname,ires,cbi+13)
+c            call pdbatm (' HZ3',resname,ires,cbi+14)
+c            call pdbatm (' HH2',resname,ires,cbi+15)
          else
             call pdbatm (' HE1',resname,ires,cbi+10)
          end if
@@ -1342,11 +1500,11 @@
             call pdbatm (' HD2',resname,ires,hca+10)
             call pdbatm (' HE1',resname,ires,hca+11)
             call pdbatm (' HE2',resname,ires,hca+12)
-         else if (biotyp(197) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+6)
-            call pdbatm (' HD2',resname,ires,cbi+7)
-            call pdbatm (' HE1',resname,ires,cbi+8)
-            call pdbatm (' HE2',resname,ires,cbi+9)
+c        else if (biotyp(197) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+6)
+c            call pdbatm (' HD2',resname,ires,cbi+7)
+c            call pdbatm (' HE1',resname,ires,cbi+8)
+c            call pdbatm (' HE2',resname,ires,cbi+9)
          else
             call pdbatm (' HD1',resname,ires,cbi+6)
             call pdbatm (' HE2',resname,ires,cbi+7)
@@ -1361,10 +1519,10 @@
             call pdbatm (' HD1',resname,ires,hca+9)
             call pdbatm (' HD2',resname,ires,hca+10)
             call pdbatm (' HE1',resname,ires,hca+11)
-         else if (biotyp(214) .ne. 0) then
-            call pdbatm (' HD1',resname,ires,cbi+6)
-            call pdbatm (' HD2',resname,ires,cbi+7)
-            call pdbatm (' HE1',resname,ires,cbi+8)
+c         else if (biotyp(214) .ne. 0) then
+c            call pdbatm (' HD1',resname,ires,cbi+6)
+c            call pdbatm (' HD2',resname,ires,cbi+7)
+c            call pdbatm (' HE1',resname,ires,cbi+8)
          else
             call pdbatm (' HD1',resname,ires,cbi+6)
          end if
@@ -1378,11 +1536,11 @@
             call pdbatm (' HD2',resname,ires,hca+9)
             call pdbatm (' HE1',resname,ires,hca+10)
             call pdbatm (' HE2',resname,ires,hca+11)
-         else if (biotyp(229) .ne. 0) then
-            call pdbatm (' HD2',resname,ires,cbi+6)
-            call pdbatm (' HE1',resname,ires,cbi+7)
-            call pdbatm (' HE2',resname,ires,cbi+8)
-         else
+c         else if (biotyp(229) .ne. 0) then
+c            call pdbatm (' HD2',resname,ires,cbi+6)
+c            call pdbatm (' HE1',resname,ires,cbi+7)
+c            call pdbatm (' HE2',resname,ires,cbi+8)
+          else
             call pdbatm (' HE2',resname,ires,cbi+6)
          end if
 c
@@ -1398,8 +1556,8 @@
 c
       else if (resname .eq. 'ASH') then
          if (allatom) then
-            call pdbatm (' HB2',resname,ires,hca+5)
-            call pdbatm (' HB3',resname,ires,hca+6)
+            call pdbatm ('1HB ',resname,ires,hca+5)
+            call pdbatm ('2HB ',resname,ires,hca+6)
             call pdbatm (' HD2',resname,ires,hca+7)
          else
             call pdbatm (' HD2',resname,ires,cbi+4)
@@ -1432,10 +1590,10 @@
 c
       else if (resname .eq. 'GLH') then
          if (allatom) then
-            call pdbatm (' HB2',resname,ires,hca+6)
-            call pdbatm (' HB3',resname,ires,hca+7)
-            call pdbatm (' HG2',resname,ires,hca+8)
-            call pdbatm (' HG3',resname,ires,hca+9)
+            call pdbatm ('1HB ',resname,ires,hca+6)
+            call pdbatm ('2HB ',resname,ires,hca+7)
+            call pdbatm ('1HG ',resname,ires,hca+8)
+            call pdbatm ('2HG ',resname,ires,hca+9)
             call pdbatm (' HE2',resname,ires,hca+10)
          else
             call pdbatm (' HE2',resname,ires,cbi+5)
@@ -1482,8 +1640,10 @@
             call pdbatm (' HE2',resname,ires,hca+12)
             call pdbatm (' HE3',resname,ires,hca+13)
             call pdbatm (' HZ1',resname,ires,hca+14)
+            if (n12(hca+5) .eq. 4) then
             call pdbatm (' HZ2',resname,ires,hca+15)
             call pdbatm (' HZ3',resname,ires,hca+16)
+            end if
          else
             call pdbatm (' HZ1',resname,ires,cbi+5)
             call pdbatm (' HZ2',resname,ires,cbi+6)
@@ -1494,19 +1654,19 @@
 c
       else if (resname .eq. 'LYD') then
          if (allatom) then
-            call pdbatm (' HB2',resname,ires,hca+6)
-            call pdbatm (' HB3',resname,ires,hca+7)
-            call pdbatm (' HG2',resname,ires,hca+8)
-            call pdbatm (' HG3',resname,ires,hca+9)
-            call pdbatm (' HD2',resname,ires,hca+10)
-            call pdbatm (' HD3',resname,ires,hca+11)
-            call pdbatm (' HE2',resname,ires,hca+12)
-            call pdbatm (' HE3',resname,ires,hca+13)
-            call pdbatm (' HZ1',resname,ires,hca+14)
-            call pdbatm (' HZ2',resname,ires,hca+15)
+            call pdbatm ('1HB ',resname,ires,hca+6)
+            call pdbatm ('2HB ',resname,ires,hca+7)
+            call pdbatm ('1HG ',resname,ires,hca+8)
+            call pdbatm ('2HG ',resname,ires,hca+9)
+            call pdbatm ('1HD ',resname,ires,hca+10)
+            call pdbatm ('2HD ',resname,ires,hca+11)
+            call pdbatm ('1HE ',resname,ires,hca+12)
+            call pdbatm ('2HE ',resname,ires,hca+13)
+            call pdbatm ('1HZ ',resname,ires,hca+14)
+            call pdbatm ('2HZ ',resname,ires,hca+15)
          else
-            call pdbatm (' HZ1',resname,ires,cbi+5)
-            call pdbatm (' HZ2',resname,ires,cbi+6)
+            call pdbatm ('1HZ ',resname,ires,cbi+5)
+            call pdbatm ('2HZ ',resname,ires,cbi+6)
          end if
 c
 c     arginine residue  (ARG)
Binary files ../../../2015_July28_serial/molcas-v8.1.15-07-21/tinker-6.3.3/source/xyzpdb.o and source/xyzpdb.o differ
